### PR TITLE
initial code for FutarchyTemplate.sol contract and tests

### PR DESCRIPTION
**Pull request by mikec**
mostly copied from https://github.com/aragon/dao-templates/blob/master/templates/company/contracts/CompanyTemplate.sol with finance, agent, and vault apps removed

### CODE REVIEW

File: contracts/FutarchyTemplate.sol [hunk 1]
```diff
@@ -1,13 +1,129 @@
...
 
+import "@aragon/templates-shared/contracts/TokenCache.sol";
 import "@aragon/templates-shared/contracts/BaseTemplate.sol";
 
-contract FutarchyTemplate is BaseTemplate {
-  constructor(DAOFactory _daoFactory, ENS _ens, MiniMeTokenFactory _miniMeFactory, IFIFSResolvingRegistrar _aragonID)
-  BaseTemplate(_daoFactory, _ens, _miniMeFactory, _aragonID)
+contract FutarchyTemplate is BaseTemplate, TokenCache {
+  string constant private ERROR_EMPTY_HOLDERS = "COMPANY_EMPTY_HOLDERS";
+  string constant private ERROR_BAD_HOLDERS_STAKES_LEN = "COMPANY_BAD_HOLDERS_STAKES_LEN";
```
Comment (ewilz) [inline contracts/FutarchyTemplate.sol:8]: company bad holders stakes len??

Comment (ewilz) [reply contracts/FutarchyTemplate.sol:8]: lol

Comment (mikec) [reply contracts/FutarchyTemplate.sol:8]: @emaG3m this is all copied from https://github.com/aragon/dao-templates/blob/master/templates/company/contracts/CompanyTemplate.sol

**Comment by ewilz** [comment]
👍

File: contracts/FutarchyTemplate.sol [hunk 2]
```diff
@@ -1,13 +1,129 @@
 pragma solidity 0.4.24;
 
+import "@aragon/templates-shared/contracts/TokenCache.sol";
 import "@aragon/templates-shared/contracts/BaseTemplate.sol";
 
-contract FutarchyTemplate is BaseTemplate {
-  constructor(DAOFactory _daoFactory, ENS _ens, MiniMeTokenFactory _miniMeFactory, IFIFSResolvingRegistrar _aragonID)
-  BaseTemplate(_daoFactory, _ens, _miniMeFactory, _aragonID)
+contract FutarchyTemplate is BaseTemplate, TokenCache {
```
Comment (ewilz) [inline contracts/FutarchyTemplate.sol:6]: where does FutarchyApp even get pulled in??

Comment (mikec) [reply contracts/FutarchyTemplate.sol:6]: @emaG3m it doesn't yet, will add that in a follow-up. just getting the base stuff in now

**Pull request merged**